sim/network.go

```diff
@@ -86,19 +87,19 @@ func (n *Network) Broadcast(msg *f3.GMessage) {
 					source:    msg.Sender,
 					dest:      k,
 					payload:   *msg,
-					deliverAt: n.clock + latency,
+					deliverAt: n.clock.Add(latency),
 				})
 		}
 	}
 }
 
 ///// Clock interface
 
-func (n *Network) Time() float64 {
-	return n.clock
+func (n *Network) Time() time.Time {
+	return n.clock.Add(time.Duration(1 * time.Millisecond)) // To ensure not exactly equal time in simulation
```

Comment:
Was this required for things to work?

Reply:
Yes, either this or placing again the `.Equal` comparisons on times

Reply:
We can have the time always moving forward by having a counter (incremented on every call to `Time()`) which adds a corresponding number of nanoseconds to the Time. This should resolve it, and it would resolve similar issues.

Reply:
Done.

Reply:
No I think that's terrible